core/src/main/java/com/exadel/aem/toolkit/core/lists/servlets/ListsServlet.java

```diff
@@ -132,15 +133,28 @@ private static boolean isList(ResourceResolver resolver, Resource resource) {
     }
 
     /**
-     * Checks whether the resource contains resources, which are either pages, or folders that may contain lists inside
+     * Checks whether the resource is a folder
      *
      * @param resource {@code Resource} instance used as the source of markup
      * @return True or false
      */
     private static boolean isFolder(Resource resource) {
+        String primaryType = resource.getValueMap().get(JcrConstants.JCR_PRIMARYTYPE, String.class);
```

Comment:
(cosmetic) If you provide a default while retrieving values ( `resource.getValueMap().get(JcrConstants.JCR_PRIMARYTYPE, StringUtils.EMPTY);` , you can omit null-check
also, [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions)  to make your code npe-safe

Reply:
That's right!
Or even:
`StringUtils.equalsAny(myString, "firstValriant", "secondVariant", "thirdValriant")`
(null check out of the box)

Reply:
Thank you for good point